thesis-fe/src/components/container/CalendarDashboard.js

```diff
@@ -1,24 +1,188 @@
+`
+
+const ContainerUsers = styled('div')`
+  margin-top: 5rem;
+  margin-left: 5rem;
+  width: 80vw;
+  height: 10vh;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  flex-wrap: wrap;
+  margin-bottom: 2rem;
+`
+const ContainerList = styled('div')`
+  max-height: 30rem;
+  width: 30rem;
+`
+
+const GoBackButton = styled('button')`
+  position: absolute;
+  right: 40vw;
+  margin-top: 2rem;
+  margin-right: .25rem;
+  position: relative;
+  font-size: 2rem;
+`;
+
+const H1 = styled('h1')`
+  font-size: 1.5rem;
+  margin: 1rem;
+  color: white;
+`;
+
+const H2 = styled('h1')`
+  font-size: 1.25rem;
+  margin: 1rem;
+  color: white;
+`;
 
 
 class MyTripsDashboard extends Component {
 
-  redirectToTrip = (id) => {
-    return () => {
-      this.props.history.push('/tripdetails/' + id)
+  state = {
+    tripID : this.props.location.pathname.split('/')[2],
+    usersInfo : {},
+    listInfo : [],
+    selectedUser :  null
+  }
+
+  generateObj = () => {
+    const objUsers = {};
+    const listPresentational = [];
+    let selected;
+    for (let i = 0; i < this.props.info.trip.timeFrame.suggestions.length; i++) {
+      if (i === 0) selected = this.props.info.trip.timeFrame.suggestions[i].creator.firstName
+      objUsers[this.props.info.trip.timeFrame.suggestions[i].creator.firstName] = this.props.info.trip.timeFrame.suggestions[i]
+      listPresentational[i] = ( <H1 onClick={this.showTimeframe} id={this.props.info.trip.timeFrame.suggestions[i].creator.firstName} key={this.props.info.trip.timeFrame.suggestions[i].creator.firstName}>
+        {this.props.info.trip.timeFrame.suggestions[i].creator.firstName}</H1> )
     }
+    this.setState({
+      usersInfo : objUsers,
+      listInfo : listPresentational,
+      selectedUser : selected
+    });
+  }
 
+  componentDidMount () {
+    this.generateObj();
   }
-  render() {
 
-    console.log(this.props);
+  redirectToTrip = () => {
+    this.props.history.push('/tripdetails/' + this.state.tripID)
+  }
 
+  showTimeframe = (e) => {
+    this.setState({
+      selectedUser : e.target.id
+    })
+  }
+  render() {
+    const mock = {};
+    mock.Arturo = ['18-12-2018','31-12-2018']
+    mock.Damien = ['12-12-2018','25-12-2018']
+    mock.Marco = ['20-12-2018','02-01-2019']
+    mock.Berta = ['17-12-2018','08-01-2019']
+    mock.Christopher = ['22-12-2018','05-01-2019']
+    const toShow = this.state.listInfo.length > 0 && (this.state.selectedUser)
     return (
-      <h1>
-        cALL
-      </h1>
+      <Container>
+        <ContainerUsers>
+        {this.state.listInfo}
+        </ContainerUsers>
+        <H2>
+          Checking {this.state.selectedUser}'s avaiability
+        </H2>
+        <DateRange
+          startDate={(toShow && mock[toShow][0]) || null}
+          endDate={(toShow && mock[toShow][1]) || null}
+          onInit={null}
+          calendars={1}
+          twoStepChange={true}
+          theme={{
```

Comment:
it would be better to keep the syles in a variable to make the code more readble